lms/djangoapps/completion_api/views.py

```diff
@@ -358,11 +364,72 @@ class CompletionDetailView(CompletionViewMixin, APIView):
     """
     # pylint: enable=line-too-long
 
-    def get(self, request, course_key):
+    def get(self, request, course_id):
         """
         Handler for GET requests.
         """
-        course_key = CourseKey.from_string(course_key)
-        progress = self.get_progress_queryset().get(course_id=course_key)
+        course_id = CourseKey.from_string(course_id)
+        try:
+            progress = self.get_progress_queryset().get(course_id=course_id)
+        except StudentProgress.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
         completion = CourseCompletionFacade(progress)
         return Response(self.get_serializer()(completion, requested_fields=self.get_requested_fields()).data)
+
+
+class CompletionBlockUpdateView(APIView):
+    """
+    API view to mark blocks as completed.
+    This is a transitional implementation.
+    """
```

Comment:
Can you document the API?  It should describe the body and URL formats, and explain what kind of responses are sent for different requests.  See the above CompletionDetailView for formatting.  This will show up in the interactive API, so it's useful for implementors.